File: 104. Maximum Depth of Binary Tree/solution.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
  int dfs(TreeNode current){
    if (current == null){
      return 0;
    }
    int leftDepth = dfs(current.left);
    int rightDepth = dfs(current.right);
    int maxDepth = leftDepth;
    if (rightDepth > maxDepth){
      maxDepth = rightDepth;
    }
    return maxDepth + 1;
  }
  public int maxDepth(TreeNode root) {
    return dfs(root);
  }
}

File: 104. Maximum Depth of Binary Tree/solution.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
  def maxDepth(self, root):
    """
    :type root: TreeNode
    :rtype: int
    """
    def dfs(current):
      if current == None:
        return 0
      leftDepth = dfs(current.left)
      rightDepth = dfs(current.right)
      maxDepth = leftDepth
      if rightDepth > maxDepth:
        maxDepth = rightDepth
      return maxDepth + 1

    return dfs(root)